### ISSUE TITLE
Question on ATEPC performance metrics and loss.

### ISSUE THREAD

**Issue by kerolzeeq** [yangheng95/PyABSA#230]
Hi author @yangheng95 ,

I'm using the FAST-LCF-ATEPC model on my custom dataset and I have 4 questions on the ATEPC performance metrics and loss:

1.  Whats the difference between these 2  Metric Visualizer (MV) tables? Is the validation set used to calculate these metrics?
![image](https://user-images.githubusercontent.com/76143604/196031328-dcc870c9-88f8-4776-a688-b04134090a4e.png)

2. As I understand from `atepc_trainer.py` , there are 3 types of losses which are `loss_ate` , `loss_apc` and lastly the combined `loss` that uses this formula `loss = loss_ate + ate_loss_weight * loss_apc`. I was wondering if you could explain it in simple terms how are each of the losses calculated from the expected output and the actual output? 

3. In continuation to question 2, I want to check if the model overfits to my dataset and to do that I need to plot the training loss and validation loss.  So does the `losses' list refer to the training loss? (see below) https://github.com/yangheng95/PyABSA/blob/964d7862da13ef8cc38cb56fe0e65086b343a9cd/pyabsa/core/atepc/training/atepc_trainer.py#L204

4.  How can I retrieve the validation loss for ATE and APC separately so that I could plot them in a graph.

Kind regards, kerolzeeq

**Comment by yangheng95** [comment]
These two MV are the same， you can refer  to  #228
The loss calculation is formulated empirically, the ate_weight are additional and not exists in the original paper.
The losses indicates the combined loss, but you can revise the code to collect loss_ate, loss_apc respectively. Then you can call plot using config.MV.losses

**Comment by kerolzeeq** [comment]
> These two MV are the same， you can refer  to  #228
> The loss calculation is formulated empirically, the ate_weight are additional and not exists in the original paper.
> The losses indicates the combined loss, but you can revise the code to collect loss_ate, loss_apc respectively. Then you can call plot using config.MV.losses
> 
> 

Thanks for the clarification, one more question.

Is there a way to get the validation loss for ate and apc instead of the training loss?

**Comment by kerolzeeq** [comment]
> > These two MV are the same， you can refer  to  #228
> > The loss calculation is formulated empirically, the ate_weight are additional and not exists in the original paper.
> > The losses indicates the combined loss, but you can revise the code to collect loss_ate, loss_apc respectively. Then you can call plot using config.MV.losses
> > 
> > 
> 
> Thanks for the clarification, one more question.
> 
> Is there a way to get the validation loss for ate and apc instead of the training loss? 

@yangheng95

**Comment by yangheng95** [comment]
Yes, technically you can, but you need to revise the evalutation part to add and save valid loss calculation:
refer to https://github.com/yangheng95/PyABSA/blob/64a917bc0ce42f83477964fbde25f7c794264ba2/pyabsa/core/atepc/training/atepc_trainer.py#L289
